src/editors/collectionDocumentsProvider.ts

```diff
@@ -14,10 +15,16 @@ export const VIEW_COLLECTION_SCHEME = 'VIEW_COLLECTION_SCHEME';
 export default class CollectionViewProvider implements vscode.TextDocumentContentProvider {
   _connectionController: ConnectionController;
   _operationsStore: CollectionDocumentsOperationsStore;
+  _statusView: StatusView;
 
-  constructor(connectionController: ConnectionController, operationsStore: CollectionDocumentsOperationsStore) {
+  constructor(
```

Comment:
When do we use constructor() and when we use activate()? What the purpose of activation?

Reply:
Ah - initially I was thinking we would create all the controllers before activation of the extension, but I think we just want to create everything in the `activate()` extension method. I'll update this in the PR: https://github.com/mongodb-js/vscode/pull/28. Just since the diff there is already big I want to avoid a large merge conflict with this PR if possible 😅

Reply:
Yeah agreed, the separate PR for it will be better! Nobody likes merge conflicts 😀

Reply:
Hmm - working on it now and I think actually separating activation of the main extension controller from the constructor might have a benefit in testing. We can have our workspace extension which has the commands registered and then create new instances of the extension which we can individually test.
I have gone and renamed the `activate` method in `connectionController` to `loadConnections` and the `activate` method in the `explorerController` to `createTreeView` to be more descriptive about what actually happens. (In https://github.com/mongodb-js/vscode/pull/28 )

Reply:
To be honest I like `activate` more...) We use activate method for compass plugins and it feels like a right naming for it. it is easier to understand what this method does if it has the same name for different controllers. and if is better for testing it is totally fine to have both activate and constructor. what do you think about continue using activate name?